### DISCUSSION TITLE
WithFaker has no documentation and doesn't seem to work as expected

### DISCUSSION THREAD

**Discussion by Patabugen** [laravel/framework#33549] [General]
- Laravel Version: 7.11
- PHP Version: 7.3.8
- Database Driver & Version: Mariadb 10.3

### Description:
I've spent today trying to use WithFaker in either my Unit or Feature tests, I've either found lots of bugs or I'm misusing it (possibly a combination)

I've been unable to find any documentation for it at all, anywhere - which is making life a bit harder. I'm not sure if I'm using it wrong, or it's broken. I'm guessing I'm using it wrong.

The main issues I'm having are:

- `setUpFaker()` doesn't accept a locale to pass to makeFaker. Meanign `setUpFaker()` can only use the default locale.
 - `makeFaker()` _always_ calls config() which doesn't seem to be available in my tests. Throwing `Target class [config] does not exist.`

If someone who knows how WithFaker should work could dump some instructions, I'm happy to write a pull request for the code/docs.

Tomorrow I will try creating a new Laravel project to see if it could be related to my project somehow.

### Steps To Reproduce:
 - Install Laravel
 - Make a unit or Feature test
 - Add `use Illuminate\Foundation\Testing\WithFaker;` to the class
 - Create a test and try to use Faker either with `$this->faker()` `$this->faker` `$this->setupFaker()` or `$this->makeFaker()`

**Comment by nandi95** [comment]
@Patabugen you would use it as `$this->faker->name`
Have you got an example repo?

**Comment by AegirLeet** [comment]
You either need to extend the `\Tests\TestCase` class or use the `\Tests\CreatesApplication` trait and call `$this->createApplication()` manually.
